python/cuml/test/test_kneighbors_regressor.py

```diff
@@ -0,0 +1,109 @@
+
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+
+from cuml.neighbors import KNeighborsRegressor as cuKNN
+
+from sklearn.datasets import make_blobs
+
+from sklearn.utils.validation import check_random_state
+from sklearn.model_selection import train_test_split
+from numpy.testing import assert_array_almost_equal
+
+import numpy as np
+
+from cuml.test.utils import array_equal
+
+
+def test_kneighbors_regressor(n_samples=40,
+                              n_features=5,
+                              n_test_pts=10,
+                              n_neighbors=3,
+                              random_state=0):
+    # Test k-neighbors regression
+    rng = np.random.RandomState(random_state)
+    X = 2 * rng.rand(n_samples, n_features) - 1
+    y = np.sqrt((X ** 2).sum(1))
+    y /= y.max()
+
+    y_target = y[:n_test_pts]
+
+    knn = cuKNN(n_neighbors=n_neighbors)
+    knn.fit(X, y)
+    epsilon = 1E-5 * (2 * rng.rand(1, n_features) - 1)
+    y_pred = knn.predict(X[:n_test_pts] + epsilon)
+    assert np.all(abs(y_pred - y_target) < 0.3)
```

Comment:
0.3 seems like a pretty high threshold... is there a reason why the deviation gets so large?

Reply:
This was copied directly from sklearn.